Mathlib/CategoryTheory/Monoidal/Cartesian/Mon_.lean

```diff
@@ -36,3 +41,191 @@ theorem lift_comp_one_right {A : C} {B : Mon_ C} (f : A ⟶ B.X) (g : A ⟶ 𝟙
+          rw [Category.assoc, ψ.mul_hom, lift_map_assoc] }
+    naturality {M N} φ := MonCat.hom_ext <| MonoidHom.ext fun f ↦ Category.assoc φ.unop f ψ.hom }
+  map_id M := NatTrans.ext <| funext fun _ ↦ MonCat.hom_ext <| MonoidHom.ext Category.comp_id
+  map_comp _ _ :=
+    NatTrans.ext <| funext fun _ ↦ MonCat.hom_ext <| MonoidHom.ext (.symm <| Category.assoc · _ _)
+
+@[reassoc]
+lemma yonedaMon_naturality (α : yonedaMonObj M ⟶ yonedaMonObj N) (f : X ⟶ Y) (g : Y ⟶ M) :
+      α.app _ (f ≫ g) = f ≫ α.app _ g := congr($(α.naturality f.op) g)
+
+variable (M) in
+/-- If `M` is a monoid object, then `Hom(-, M)` as a presheaf of monoids is represented by `M`. -/
+def yonedaMonObjRepresentableBy : (yonedaMonObj M ⋙ forget _).RepresentableBy M :=
+  Functor.representableByEquiv.symm (.refl _)
+
+variable (M) in
+lemma Mon_Class.ofRepresentableBy_yonedaMonObjRepresentableBy :
+    ofRepresentableBy M _ (yonedaMonObjRepresentableBy M) = ‹_› := by
+  ext; show lift (fst M M) (snd M M) ≫ μ = μ; rw [lift_fst_snd, Category.id_comp]
+
+@[deprecated (since := "2025-03-07")]
+alias Mon_ClassOfRepresentableBy_yonedaMonObjRepresentableBy :=
+  Mon_Class.ofRepresentableBy_yonedaMonObjRepresentableBy
+
+/-- The yoneda embedding for `Mon_C` is fully faithful. -/
+def yonedaMonFullyFaithful : yonedaMon (C := C).FullyFaithful where
+  preimage {M N} α :=
+  { hom := α.app (op M.X) (𝟙 M.X)
+    one_hom := by
+      dsimp only [yonedaMon_obj] at α ⊢
+      rw [← yonedaMon_naturality, Category.comp_id,
+        ← Category.id_comp M.one, toUnit_unique (𝟙 _) (toUnit _),
+        ← Category.id_comp N.one, toUnit_unique (𝟙 _) (toUnit _)]
+      exact (α.app _).hom.map_one
+    mul_hom := by
+      dsimp only [yonedaMon_obj] at α ⊢
+      rw [← yonedaMon_naturality, Category.comp_id, ← Category.id_comp M.mul, ← lift_fst_snd]
+      refine ((α.app _).hom.map_mul _ _).trans ?_
+      show lift _ _ ≫ N.mul = _
+      congr 1
+      ext <;> simp only [lift_fst, tensorHom_fst, lift_snd, tensorHom_snd,
+        ← yonedaMon_naturality, Category.comp_id] }
+  map_preimage {M N} α := by
+    ext Y f
+    dsimp only [yonedaMon_obj, yonedaMon_map_app, MonCat.hom_ofHom]
+    simp_rw [← yonedaMon_naturality]
+    simp
+  preimage_map φ := Mon_.ext (Category.id_comp φ.hom)
+
+instance : yonedaMon (C := C).Full := yonedaMonFullyFaithful.full
+instance : yonedaMon (C := C).Faithful := yonedaMonFullyFaithful.faithful
+
+lemma essImage_yonedaMon :
+    yonedaMon (C := C).essImage = (· ⋙ forget _) ⁻¹' setOf Functor.IsRepresentable := by
+  ext F
+  constructor
+  · rintro ⟨M, ⟨α⟩⟩
+    exact ⟨M.X, ⟨Functor.representableByEquiv.symm (isoWhiskerRight α (forget _))⟩⟩
+  · rintro ⟨X, ⟨e⟩⟩
+    letI := Mon_Class.ofRepresentableBy X F e
+    exact ⟨.mk' X, ⟨yonedaMonObjEquivOfRepresentableBy X F e⟩⟩
+
+@[reassoc]
+lemma Mon_Class.comp_mul (f : X ⟶ Y) (g h : Y ⟶ M) : f ≫ (g * h) = f ≫ g * f ≫ h :=
+  ((yonedaMon.obj <| .mk' M).map f.op).hom.map_mul g h
```

Comment:
Can you similarly provide `mul_comp` (when the composed map `IsMon_Hom`) and `one_comp` and `comp_one`? And also perhaps `comp_inv` and `inv_comp` for groups.